iroh/src/endpoint.rs

```diff
@@ -132,7 +131,7 @@ pub struct Builder {
     transport_config: quinn::TransportConfig,
     keylog: bool,
     #[debug(skip)]
-    discovery: Vec<DiscoveryBuilder>,
+    discovery: Vec<Box<dyn DynIntoDiscovery>>,
```

Comment:
Sorry for the drive-by comment, but looking at the `#[debug(skip)]` above: We could consider making `DynIntoDiscovery: std::fmt::Debug`, so that we don't have to do this (and get debug output if needed in the future).

Also, it's a bound that's easier to remove than to add in the future.

But that's really just a nit at best.